presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockEncodingBuffers.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import io.airlift.slice.ByteArrays;
+import io.airlift.slice.SliceOutput;
+
+import static com.facebook.presto.spi.block.ByteArrayUtils.writeLengthPrefixedString;
+import static com.facebook.presto.spi.block.ColumnarArray.toColumnarArray;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static java.lang.Math.max;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
+public class ArrayBlockEncodingBuffers
+        extends BlockEncodingBuffers
+{
+    public static final String NAME = "LONG_ARRAY";
+    public static final int DEFAULT_MAX_ELEMENT_COUNT = 512 * 1024;
+
+    // TODO: These are single piece buffers for now. They will become linked list of pages requested from buffer pool.
+    // TODO: we need to decide how we can operate on the buffers. Slice itself doesn't allow appending, and DynamicSliceOutput does capacity check for every value. If we modify DynamicSliceOutput we need to modify Airlift.
+    // If we directly operate on byte[] we need to decide the interface. Currently we'll use ByteArrayUtils for updating batch of rows, and ByteArrays in Airlift to update single value
+    private byte[] nullsBuffer;
+    private int nullsBufferIndex;  //The next byte address if new values to be added.
+    private byte[] offsetsBuffer;
+    private int offsetsBufferIndex;  //The next byte address if new offset to be added.
+    private BlockEncodingBuffers rawBlockBuffer;
```

Comment:
I can't find a place that initializes rawBlockBuffer. Would you give me a pointer?

Reply:
Oops I removed it and forgot to add it back. Thanks for catching.